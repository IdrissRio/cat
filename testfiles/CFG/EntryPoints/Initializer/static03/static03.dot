digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "StaticInitializer[28]"; "FieldDecl[42]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Block[29]"; "Modifiers[43]"; "PrimitiveTypeAccess[46]"; "List[47]"}
{ rank = same; "List[30]"; "List[44]"; "FieldDeclarator[48]"}
{ rank = same; "ExprStmt[31]"; "Modifier[45]"; "List[49]"; "Opt[50]"}
{ rank = same; "Dot[32]"; "IntegerLiteral[51]"}
{ rank = same; "Dot[33]"; "MethodAccess[36]"}
{ rank = same; "TypeAccess[34]"; "VarAccess[35]"; "List[38]"; "List[40]"; "List[41]"; "UncheckedExceptions[37]"}
{ rank = same; "StringLiteral[39]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < static03 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StaticInitializer[28]" [label="StaticInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[29]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[31]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[32]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[33]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[34]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[35]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[36]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[39]" [label="StringLiteral\n < test >", fillcolor=white   style=filled  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[41]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[37]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDecl[42]" [label="FieldDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[43]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[45]" [label="Modifier\n < static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[46]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[47]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDeclarator[48]" [label="FieldDeclarator\n < i >", fillcolor=white   style=filled  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[50]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[51]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"Dot[33]" -> "TypeAccess[34]" [style=dashed, color=gray]
"Dot[33]" -> "VarAccess[35]" [style=dashed, color=gray]
"Dot[32]" -> "Dot[33]" [style=dashed, color=gray]
"List[38]" -> "StringLiteral[39]" [style=dashed, color=gray]
"MethodAccess[36]" -> "List[38]" [style=dashed, color=gray]
"MethodAccess[36]" -> "List[40]" [style=dashed, color=gray]
"MethodAccess[36]" -> "List[41]" [style=dashed, color=gray]
"MethodAccess[36]" -> "UncheckedExceptions[37]" [style=dashed, color=gray]
"Dot[32]" -> "MethodAccess[36]" [style=dashed, color=gray]
"ExprStmt[31]" -> "Dot[32]" [style=dashed, color=gray]
"List[30]" -> "ExprStmt[31]" [style=dashed, color=gray]
"Block[29]" -> "List[30]" [style=dashed, color=gray]
"StaticInitializer[28]" -> "Block[29]" [style=dashed, color=gray]
"List[27]" -> "StaticInitializer[28]" [style=dashed, color=gray]
"List[44]" -> "Modifier[45]" [style=dashed, color=gray]
"Modifiers[43]" -> "List[44]" [style=dashed, color=gray]
"FieldDecl[42]" -> "Modifiers[43]" [style=dashed, color=gray]
"FieldDecl[42]" -> "PrimitiveTypeAccess[46]" [style=dashed, color=gray]
"FieldDeclarator[48]" -> "List[49]" [style=dashed, color=gray]
"Opt[50]" -> "IntegerLiteral[51]" [style=dashed, color=gray]
"FieldDeclarator[48]" -> "Opt[50]" [style=dashed, color=gray]
"List[47]" -> "FieldDeclarator[48]" [style=dashed, color=gray]
"FieldDecl[42]" -> "List[47]" [style=dashed, color=gray]
"List[27]" -> "FieldDecl[42]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "FieldDeclarator[48]" [color=red, constraint=false]
"FieldDeclarator[48]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"FieldDeclarator[48]" -> "IntegerLiteral[51]" [color=red, constraint=false]
"IntegerLiteral[51]" -> "FieldDeclarator[48]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[51]" -> "MethodAccess[36]" [color=red, constraint=false]
"MethodAccess[36]" -> "IntegerLiteral[51]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[36]" -> "StringLiteral[39]" [color=red, constraint=false]
"StringLiteral[39]" -> "MethodAccess[36]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[39]" -> "VarAccess[35]" [color=red, constraint=false]
"VarAccess[35]" -> "StringLiteral[39]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[35]" -> "TypeAccess[34]" [color=red, constraint=false]
"TypeAccess[34]" -> "VarAccess[35]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[34]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "TypeAccess[34]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"StaticInitializer[28]" -> "FieldDecl[42]" [style=invis]
"Dot[33]" -> "MethodAccess[36]" [style=invis]
"TypeAccess[34]" -> "VarAccess[35]" [style=invis]
"Modifiers[43]" -> "PrimitiveTypeAccess[46]" -> "List[47]" [style=invis]
"List[49]" -> "Opt[50]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
42", fillcolor=1]
Edges [label="Total Nr. Edges
57", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
8", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}