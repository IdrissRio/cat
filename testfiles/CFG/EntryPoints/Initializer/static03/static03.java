public class static03 {
  static { System.out.println("test"); }
  static int i = 0;
}