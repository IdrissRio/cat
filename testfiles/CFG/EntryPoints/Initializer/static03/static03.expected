NodeName: Entry[15] Successors: [TypeAccess[34]]
NodeName: TypeAccess[34] Successors: [VarAccess[35]]
NodeName: VarAccess[35] Successors: [StringLiteral[39]]
NodeName: StringLiteral[39] Successors: [MethodAccess[36]]
NodeName: MethodAccess[36] Successors: [IntegerLiteral[51]]
NodeName: IntegerLiteral[51] Successors: [FieldDeclarator[48]]
NodeName: FieldDeclarator[48] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []

NodeName: Exit[16] Predecessor: [FieldDeclarator[48]]
NodeName: FieldDeclarator[48] Predecessor: [IntegerLiteral[51]]
NodeName: IntegerLiteral[51] Predecessor: [MethodAccess[36]]
NodeName: MethodAccess[36] Predecessor: [StringLiteral[39]]
NodeName: StringLiteral[39] Predecessor: [VarAccess[35]]
NodeName: VarAccess[35] Predecessor: [TypeAccess[34]]
NodeName: TypeAccess[34] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
